### PR TITLE
M m leapset fix

### PR DESCRIPTION
**Pull request by particlebanana**
If I understand #928 correctly then this should give the same result without the need for the transformation.

### CODE REVIEW

**Comment by dmarcelino** [comment]
I like this! :white_check_mark:

File: lib/waterline/query/dql/destroy.js [hunk 1]
```diff
@@ -88,7 +93,14 @@ module.exports = function(criteria, cb) {
         // than crashing.
         if(!refKey) return next();
 
-        var mappedValues = result.map(function(vals) { return vals[pk]; });
+        // Make sure we don't return any undefined pks
+        var mappedValues = result.reduce(function(memo, vals) {
+          if (vals[pk] !== undefined) {
```
Comment (dmarcelino) [inline lib/waterline/query/dql/destroy.js]: Are undefined pks legitimate?

**Comment by dmarcelino** [comment]
One question that @devinivy has asked in https://github.com/balderdashy/waterline/pull/928#issuecomment-93990300: is it legitimate to have undefined pas? Or should waterline throw an error there?

**Comment by particlebanana** [comment]
I can't think of a situation where you would get a record back that didn't have a primary key. The only way this should happen would be if you were using `columnName` in the `map` which this PR should fix.

Anyone have any ideas how/why it would happen?

File: lib/waterline/query/dql/destroy.js (hunk 1 above)
Comment (tjwebb) [inline lib/waterline/query/dql/destroy.js]: No

Comment (dmarcelino) [inline lib/waterline/query/dql/destroy.js]: So we probably don't need this code, I guess..

**Comment by esopian** [comment]
Looks good to me.  :+1:

**Pull request merged**